### PR TITLE
adds `use_ccby_license()` to add a CCBY 4.0 license (#547)

### LINKED ISSUE #547: feature request: use_ccby_license()
Hello!

I was wondering if there might be interest in a `use_ccby_license()` function?

Happy to submit a PR, if that is of interest.

Cheers,

Nick

### PR DESCRIPTION
**Pull request by njtierney**
This resolves #547

### CODE REVIEW

File: R/license.R [hunk 1]
```diff
@@ -87,6 +87,18 @@ use_cc0_license <- function(name = find_name()) {
...
 
+#' @rdname licenses
+#' @export
+use_ccby_license <- function(name = find_name()) {
+  force(name)
+  check_is_package("use_ccby_license()")
+
+  use_description_field("License",
+                        "CCBY-4.0",
+                        overwrite = TRUE)
```
Comment (jennybc) [inline R/license.R]: Can you use styler on the bits you're changing? Re: indentation and line breaks, specifically.

Comment (njtierney) [reply R/license.R]: I did this with the styler adding, but it looks like the style is inconsistent with other functions in license.R - did you want me to run it on the whole license.R file?

Comment (jennybc) [reply R/license.R]: Sure it's fine to style this whole file.

File: NEWS.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 ## New features
 
+* `use_ccby_license()` adds a CCBY 4.0 license (#547)
```
Comment (jennybc) [inline NEWS.md]: Also put your GitHub username here (look around NEWS for examples).

**Comment by njtierney** [comment]
OK, done!

**Pull request merged**